pkg/capacity/planner.go

```diff
@@ -554,3 +559,23 @@ func (p *NaivePlanner) handlePoolExpiration(cancelOld bool) error {
 
 	return nil
 }
+
+// usesExpiredResources checks if a workload uses expired resources in the pool.
+//
+// No checks are done to make sure the workload is actually part of the pool. This
+// method only checks if a workload is using a non zero amount of a resource which
+// is no longer present in the pool.
+func usesExpiredResources(pool types.Pool, workload workloads.Workloader) bool {
+	// Only delete workloads we actually need to delete
+	// In other words, only delete the workload if it consumes resources
+	// which are empty
+
+	cu, su := CloudUnitsFromResourceUnits(workload.GetRSU())
+	if cu > 0 && pool.Cus == 0 {
```

Comment:
I guess we ensure that pool CU and SU cannot be negative right ?
otherwise we can protect it by using `<=0` instead of `==0`

Reply:
the pools are synced when they are loaded, since that needs to happen anyways, and in doing so negative values are clamped to 0 indeed. Added it though since it should not matter performance wise